File: client/src/app/admin-ui/models/school-user.model.ts
import {Roles} from "../../shared/enums/school-user-roles";
import {School} from "../../shared/models/school.interface";

export class SchoolUserInputModel {
    id: string;
    firstName: string;
    secondName: string;
    lastName: string;
    pin: string;
    address: string;
    town: string;
    schoolId: string;
    role: Roles;
}

File: server/BusinessLogicLayer/DTOs/Models/SchoolUserModels/SchoolUserModel.cs
using System.ComponentModel.DataAnnotations;

namespace SchoolBook.BusinessLogicLayer.DTOs.Models.SchoolUserModels
{
    public class SchoolUserModel
    {
        [Required] 
        public string FirstName { get; set; }

        public string SecondName { get; set; }

        [Required] 
        public string LastName { get; set; }

        [Required] 
        public string Pin { get; set; }

        [Required] 
        public string Address { get; set; }

        [Required]
        public string Town { get; set; }

        [Required] 
        public string SchoolId { get; set; }
    }
}
